src/bridge.rs

```diff
@@ -11,6 +11,29 @@ pub struct ResourceIdentifier {
     pub rtype: String,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Device {
+    pub id: String,
+    pub id_v1: Option<String>,
+    pub services: Vec<ResourceIdentifier>,
+}
+
+impl Device {
+    /// Returns the ids of all services of type light associated with this device.
+    pub fn get_lights(&self) -> Vec<String> {
+        self.services
+            .iter()
+            .filter_map(|service| {
+                if service.rtype == "light" {
+                    Some(service.rid.clone())
+                } else {
+                    None
+                }
+            })
+            .collect()
```

Comment:
ca, we return an impl Iterator<&str> here ? i feel it would avoid cloning strings and allocating a vec...